### DISCUSSION TITLE
How/where does Sdk.SuppressInstrumentation get set?

### DISCUSSION THREAD

**Discussion by jamescrosswell** [open-telemetry/opentelemetry-dotnet#5344] [Q&A]
I've got an application that is instrumented with OpenTelemetry (it's an Aspire starter template) and I'm trying to capture trace information via a custom span process. One of the things the span processor does is check the value of `Sdk.SuppressInstrumentation`... if it's true then the span processor bails rather than forward the traces. 

For some reason, somewhere/somehow in that Aspire starter template, `Sdk.SuppressInstrumentation` is being set to `true` and I wondered why this would be the case? What's the purpose of `Sdk.SuppressInstrumentation` and what are the standard mechanisms to enable or disable this?

**Comment by jamescrosswell** [comment]
Hm, looks like it's set for logging, metrics and traces... all for the same reason:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3157b5ce12acd14cbd4e85dac01e7c3a283a3caa/docs/logs/extending-the-sdk/MyExporter.cs#L32-L34

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3157b5ce12acd14cbd4e85dac01e7c3a283a3caa/docs/metrics/extending-the-sdk/MyExporter.cs#L32-L34

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3157b5ce12acd14cbd4e85dac01e7c3a283a3caa/docs/trace/extending-the-sdk/MyExporter.cs#L32-L34

So something to do with the exporters. Is it possible to capture telemetry via a custom span processor if exporters are also configured for the telemetry data or are those two things fundamentally incompatible?

**Reply by cijothomas** [reply]
Not clear what you are trying to achieve...Could you elaborate on that, so we can help?

(The reason for custom exporters doing supress is already in the doc comment, right? Do we need more clarification on that?)

**Reply by jamescrosswell** [reply]
Yes I think I found the reason by cracking open the source code. I didn't/couldn't find it anywhere in the docs (not sure if that's a me problem or a docs problem). 

Originally I was concerned about this because I was seeing a bunch of log messages from Sentry:
https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry.OpenTelemetry/SentryPropagator.cs#L89-L94

However the telemetry from Aspire is actually getting through to Sentry so I suspect what I'm seeing here is normal behaviour. There's some telemetry from the exporters that is being suppressed to prevent a live-loop and that's showing in our logs... but the other (normal) telemetry events are being handled as we would want/expect.
